src/test/java/seedu/sellsavvy/model/order/DateTest.java

```diff
@@ -20,24 +22,52 @@ public void constructor_invalidDate_throwsIllegalArgumentException() {
     }
 
     @Test
-    public void isValidDate() {
+    public void isValidDateRegex() {
         // null date value
-        assertThrows(NullPointerException.class, () -> Date.isValidDate(null));
+        assertThrows(NullPointerException.class, () -> Date.isValidDateRegex(null));
+
+        // invalid date regex
+        assertFalse(Date.isValidDateRegex("")); // empty string
+        assertFalse(Date.isValidDateRegex(" ")); // spaces only
+        assertFalse(Date.isValidDateRegex("date")); // not a date
+        assertFalse(Date.isValidDateRegex("2-02-2002")); // invalid day format
+        assertFalse(Date.isValidDateRegex("02/02/2002")); // non-hyphen non-numeric characters
+        assertFalse(Date.isValidDateRegex("02-22-2002")); // invalid month
+        assertFalse(Date.isValidDateRegex("02-02-2222")); // not within 21st century
```

Comment:
Can there be another test case for 20th century or earlier? That EP isn't covered here.

Reply:
Updated in commit.